app/services/LocationService.js

```diff
@@ -105,278 +105,223 @@ export class LocationData {
-              message: languages.t('label.location_disabled_message'),
-            });
-          } else {
-            PushNotification.cancelLocalNotifications({
-              id: LOCATION_DISABLED_NOTIFICATION,
-            });
-          }
-        });
-      }
-    });
+  BackgroundGeolocation.on('foreground', () => {
+    console.log('[INFO] App is in foreground');
+  });
 
-    BackgroundGeolocation.on('background', () => {
-      console.log('[INFO] App is in background');
-    });
+  BackgroundGeolocation.on('abort_requested', () => {
+    console.log('[INFO] Server responded with 285 Updates Not Required');
+    // Here we can decide whether we want stop the updates or not.
+    // If you've configured the server to return 285, then it means the server does not require further update.
+    // So the normal thing to do here would be to `BackgroundGeolocation.stop()`.
+    // But you might be counting on it to receive location updates in the UI, so you could just reconfigure and set `url` to null.
+  });
 
-    BackgroundGeolocation.on('foreground', () => {
-      console.log('[INFO] App is in foreground');
-    });
+  BackgroundGeolocation.on('http_authorization', () => {
+    console.log('[INFO] App needs to authorize the http requests');
+  });
 
-    BackgroundGeolocation.on('abort_requested', () => {
-      console.log('[INFO] Server responded with 285 Updates Not Required');
-      // Here we can decide whether we want stop the updates or not.
-      // If you've configured the server to return 285, then it means the server does not require further update.
-      // So the normal thing to do here would be to `BackgroundGeolocation.stop()`.
-      // But you might be counting on it to receive location updates in the UI, so you could just reconfigure and set `url` to null.
+  BackgroundGeolocation.on('stop', () => {
+    PushNotification.localNotification({
+      title: languages.t('label.location_disabled_title'),
+      message: languages.t('label.location_disabled_message'),
     });
+    console.log('[INFO] stop');
+  });
+  BackgroundGeolocation.on('stationary', () => {
+    console.log('[INFO] stationary');
+  });
+}
 
-    BackgroundGeolocation.on('http_authorization', () => {
-      console.log('[INFO] App needs to authorize the http requests');
-    });
+export default class LocationServices {
+  static async start() {
+    // handles edge cases around Android where start might get called again even though
+    // the service is already created.  Make sure the listeners are still bound and exit
+    if (isBackgroundGeolocationConfigured) {
+      BackgroundGeolocation.start();
+      return;
+    }
 
-    BackgroundGeolocation.on('stop', () => {
-      PushNotification.localNotification({
-        title: languages.t('label.location_disabled_title'),
-        message: languages.t('label.location_disabled_message'),
-      });
-      console.log('[INFO] stop');
-    });
+    configureBackroundGeoService();
```

Comment:
The diffs are a bit off, but I simply moved all the code here into a method. I did not change it at all.

It was clogging up the general overview of this method which is to configure once + start every time.

Reply:
I removed this because it made the diff harder to parse